File: integration/android/LoginHostScreen.kt
import androidx.compose.runtime.Composable

class LoginHostStore {
    var screenState = LoginScreenState()

    fun onEmailChanged(value: String) {
        screenState = screenState.copy(email = value)
    }

    fun onPasswordChanged(value: String) {
        screenState = screenState.copy(password = value)
    }

    fun submitLogin() {}
}

@Composable
fun LoginHostScreen(store: LoginHostStore = LoginHostStore()) {
    LoginScreen(
        state = store.screenState,
        actions = LoginScreenActions(
            onEmailChanged = { value -> store.onEmailChanged(value) },
            onPasswordChanged = { value -> store.onPasswordChanged(value) },
            onSubmitLogin = { store.submitLogin() }
        )
    )
}

File: integration/ios/LoginHostScreen.swift
import SwiftUI

final class LoginHostStore {
    func submitLogin() {}
}

struct LoginHostScreen: View {
    @State private var screenState = LoginScreenState()
    private let store = LoginHostStore()

    var body: some View {
        LoginScreen(
            state: $screenState,
            actions: LoginScreenActions(
                submitLogin: { store.submitLogin() }
            )
        )
    }
}
